src/main/scala/common/Syntax.scala

```diff
@@ -70,8 +73,11 @@ object Syntax {
   // Use case class instead of case object to get unique positions
   case class TVoid() extends Type
   case class TBool() extends Type
+
+  case class TRational(value:Double) extends Type 
```

Comment:
This is a pretty fine-grained thing and probably doesn't matter too much, but we probably want to represent the underlying value precisely—i.e., in the same way that `ERational` represents the number, which is currently a string. (You could also imagine representing it as a pair of integers, the integer and fractional parts, but a string is basically just as good.)

Reply:
Done